lib/gui/app/modules/analytics.ts

```diff
@@ -15,108 +15,75 @@
-	resinCorvus.install({
-		services: {
-			sentry: sentryToken,
-			mixpanel: mixpanelToken,
-		},
-		options: {
-			release: packageJSON.version,
-			shouldReport: () => {
-				return settings.getSync('errorReporting');
-			},
-			mixpanelDeferred: true,
-		},
-	});
-}
-
-let mixpanelSample = DEFAULT_PROBABILITY;
-
+let analyticsClient: Client;
 /**
  * @summary Init analytics configurations
  */
-async function initConfig() {
-	await installCorvus();
-	let validatedConfig = null;
-	try {
-		const configUrl = await settings.get('configUrl');
-		const config = await getConfig(configUrl);
-		const mixpanel = _.get(config, ['analytics', 'mixpanel'], {});
-		mixpanelSample = mixpanel.probability || DEFAULT_PROBABILITY;
-		if (isClientEligible(mixpanelSample)) {
-			validatedConfig = validateMixpanelConfig(mixpanel);
-		}
-	} catch (err) {
-		resinCorvus.logException(err);
+export const initAnalytics = _.once(async () => {
+	const dsn =
+		(await settings.get('analyticsSentryToken')) ||
+		_.get(packageJSON, ['analytics', 'sentry', 'token']);
+	if (dsn) {
+		SentryRenderer.init({ dsn });
 	}
-	resinCorvus.setConfigs({
-		mixpanel: validatedConfig,
-	});
-}
-
-initConfig();
 
-/**
- * @summary Check that the client is eligible for analytics
- */
-function isClientEligible(probability: number) {
-	return Math.random() < probability;
-}
+	const projectName =
+		(await settings.get('analyticsAmplitudeToken')) ||
+		_.get(packageJSON, ['analytics', 'amplitude', 'token']);
 
-/**
- * @summary Check that config has at least HTTP_PROTOCOL and api_host
- */
-function validateMixpanelConfig(config: {
-	api_host?: string;
-	HTTP_PROTOCOL?: string;
-}) {
-	const mixpanelConfig = {
-		api_host: 'https://api.mixpanel.com',
+	(global as any).fetch = fetch;
+	const clientConfig = {
+		projectName,
+		endpoint: 'data.balena-staging.com',
+		componentName: 'etcher',
+		componentVersion: packageJSON.version,
 	};
-	if (config.HTTP_PROTOCOL !== undefined && config.api_host !== undefined) {
-		mixpanelConfig.api_host = `${config.HTTP_PROTOCOL}://${config.api_host}`;
-	}
-	return mixpanelConfig;
-}
+	analyticsClient = projectName
+		? createClient(clientConfig)
+		: createNoopClient();
+});
 
-/**
- * @summary Log an event
- *
- * @description
- * This function sends the debug message to product analytics services.
- */
-export function logEvent(message: string, data: _.Dictionary<any> = {}) {
+function reportAnalytics(message: string, data: _.Dictionary<any> = {}) {
 	const { applicationSessionUuid, flashingWorkflowUuid } = store
 		.getState()
 		.toJS();
-	resinCorvus.logEvent(message, {
+
+	analyticsClient.track(message, {
 		...data,
-		sample: mixpanelSample,
 		applicationSessionUuid,
 		flashingWorkflowUuid,
 	});
 }
 
+/**
+ * @summary Log an event
+ *
+ * @description
+ * This function sends the debug message to product analytics services.
+ */
+export async function logEvent(message: string, data: _.Dictionary<any> = {}) {
+	await initAnalytics();
```

Comment:
Would there be any reason to not move `initAnalytics` to _inside_ the `if (shouldReportAnalytics)` check?